### PR TITLE
[Metrics SDK] Change Prometheus CMake target name

### LINKED ISSUE #1764: [Metrics SDK] Inconsistent CMake target name for Prometheus exporter
Prometheus exporter cmake target name is not consistent with other otel-cpp targets:

```bash
$ ls -ltr /usr/local/lib/libopentelemetry_* /usr/local/lib/libprometheus_*
-rw-r--r-- 1 root root     3300 Nov  3 10:31 /usr/local/lib/libopentelemetry_version.a
-rw-r--r-- 1 root root   144864 Nov  3 10:31 /usr/local/lib/libopentelemetry_common.a
-rw-r--r-- 1 root root  1910868 Nov  3 10:31 /usr/local/lib/libopentelemetry_http_client_curl.a
-rw-r--r-- 1 root root  2794762 Nov  3 10:31 /usr/local/lib/libopentelemetry_resources.a
-rw-r--r-- 1 root root  5777132 Nov  3 10:31 /usr/local/lib/libopentelemetry_trace.a
-rw-r--r-- 1 root root 19872902 Nov  3 10:31 /usr/local/lib/libopentelemetry_metrics.a
-rw-r--r-- 1 root root  2444236 Nov  3 10:31 /usr/local/lib/libopentelemetry_exporter_in_memory.a
-rw-r--r-- 1 root root  2887480 Nov  3 10:31 /usr/local/lib/libopentelemetry_exporter_ostream_span.a
-rw-r--r-- 1 root root  1559756 Nov  3 10:31 /usr/local/lib/libopentelemetry_exporter_ostream_metrics.a
-rw-r--r-- 1 root root  2175544 Nov 10 16:15 /usr/local/lib/libprometheus_exporter.a
-rw-r--r-- 1 root root  4246384 Nov 10 16:17 /usr/local/lib/libopentelemetry_exporter_zipkin_trace.a

```

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #1764

Changed the target name to `opentelemetry_exporter_prometheus` .

```bash
$ ls /usr/local/lib/libopentelemetry_* -ltr
-rw-r--r-- 1 root root  5884858 Oct 28 08:10 /usr/local/lib/libopentelemetry_proto.a
-rw-r--r-- 1 root root   874006 Oct 28 16:19 /usr/local/lib/libopentelemetry_otlp_recordable.a
-rw-r--r-- 1 root root     3300 Nov 10 16:11 /usr/local/lib/libopentelemetry_version.a
-rw-r--r-- 1 root root   144904 Nov 10 16:11 /usr/local/lib/libopentelemetry_common.a
-rw-r--r-- 1 root root  2794826 Nov 10 16:11 /usr/local/lib/libopentelemetry_resources.a
-rw-r--r-- 1 root root  1910868 Nov 10 16:11 /usr/local/lib/libopentelemetry_http_client_curl.a
-rw-r--r-- 1 root root  5777388 Nov 10 16:11 /usr/local/lib/libopentelemetry_trace.a
-rw-r--r-- 1 root root 19873766 Nov 10 16:11 /usr/local/lib/libopentelemetry_metrics.a
-rw-r--r-- 1 root root  1559820 Nov 10 16:13 /usr/local/lib/libopentelemetry_exporter_ostream_metrics.a
-rw-r--r-- 1 root root  2175672 Nov 10 16:13 /usr/local/lib/libopentelemetry_exporter_prometheus.a
-rw-r--r-- 1 root root  2887576 Nov 10 16:13 /usr/local/lib/libopentelemetry_exporter_ostream_span.a
-rw-r--r-- 1 root root  2444300 Nov 10 16:13 /usr/local/lib/libopentelemetry_exporter_in_memory.a
-rw-r--r-- 1 root root   411284 Nov 10 16:24 /usr/local/lib/libopentelemetry_exporter_otlp_grpc_client.a
-rw-r--r-- 1 root root     2518 Nov 10 16:24 /usr/local/lib/libopentelemetry_exporter_otlp_grpc_log.a
-rw-r--r-- 1 root root  1780364 Nov 10 16:24 /usr/local/lib/libopentelemetry_exporter_otlp_http_client.a
-rw-r--r-- 1 root root   941658 Nov 10 16:24 /usr/local/lib/libopentelemetry_exporter_otlp_http_metric.a
-rw-r--r-- 1 root root  1017242 Nov 10 16:24 /usr/local/lib/libopentelemetry_exporter_otlp_http.a
-rw-r--r-- 1 root root  1021982 Nov 10 16:24 /usr/local/lib/libopentelemetry_exporter_otlp_grpc_metrics.a
-rw-r--r-- 1 root root  1093522 Nov 10 16:24 /usr/local/lib/libopentelemetry_exporter_otlp_grpc.a
-rw-r--r-- 1 root root  4246512 Nov 10 16:29 /usr/local/lib/libopentelemetry_exporter_zipkin_trace.a
```

### CODE REVIEW

File: exporters/prometheus/CMakeLists.txt [hunk 1]
```diff
@@ -17,23 +17,23 @@ if(NOT TARGET prometheus-cpp::core)
   find_package(prometheus-cpp CONFIG REQUIRED)
 endif()
 
-add_library(prometheus_exporter src/exporter.cc src/collector.cc
-                                src/exporter_utils.cc)
+add_library(opentelemetry_exporter_prometheus src/exporter.cc src/collector.cc
+                                              src/exporter_utils.cc)
 target_include_directories(
-  prometheus_exporter
+  opetelemetry_exporter_prometheus
```
Comment (marcalff) [inline exporters/prometheus/CMakeLists.txt:23]: Typo.

opetelemetry_exporter_prometheus - > opentelemetry_exporter_prometheus

File: exporters/prometheus/CMakeLists.txt [hunk 2]
```diff
@@ -17,23 +17,23 @@ if(NOT TARGET prometheus-cpp::core)
...
   PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
          "$<INSTALL_INTERFACE:include>")
 
-set(PROMETHEUS_EXPORTER_TARGETS prometheus_exporter)
+set(PROMETHEUS_EXPORTER_TARGETS opetelemetry_exporter_prometheus)
```
Comment (marcalff) [inline exporters/prometheus/CMakeLists.txt:27]: Same typo

File: exporters/prometheus/CMakeLists.txt [hunk 3]
```diff
@@ -17,23 +17,23 @@ if(NOT TARGET prometheus-cpp::core)
...
 if(TARGET pull)
   list(APPEND PROMETHEUS_EXPORTER_TARGETS pull)
 endif()
 if(TARGET core)
   list(APPEND PROMETHEUS_EXPORTER_TARGETS core)
 endif()
 target_link_libraries(
-  prometheus_exporter PUBLIC opentelemetry_metrics prometheus-cpp::pull
-                             prometheus-cpp::core)
+  opetelemetry_exporter_prometheus
```
Comment (marcalff) [inline exporters/prometheus/CMakeLists.txt:35]: Same typo

**Review by marcalff — CHANGES_REQUESTED** [review]

File: exporters/prometheus/CMakeLists.txt (hunk 1 above)
Comment (lalitb) [reply exporters/prometheus/CMakeLists.txt]: Thanks. forgot to push all the changes after testing them locally. Have fixed now.

Comment (marcalff) [reply exporters/prometheus/CMakeLists.txt]: Thanks, resolved then.

**Review by marcalff — APPROVED** [review]
LGTM, with format fix

**Review by owent — APPROVED** [review]
LGTM

**Review by esigo — APPROVED** [review]
LGTM
Thanks for the fix :)

**Comment by esigo** [comment]
I think we need to clean cache: 
```
FATAL: corrupt installation: file '/Users/runner/.cache/bazel/install/055c20a5e[6](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/3441965197/jobs/5762053586#step:4:7)23d48b44a7c0f17b28c93a/A-server.jar' is missing or modified.  Please remove '/Users/runner/.cache/bazel/install/055c20a5e623d48b44a7c0f17b28c93a' and try again.
Error: Process completed with exit code 36.
```

**Comment by lalitb** [comment]
> I think we need to clean cache:
> 
> ```
> FATAL: corrupt installation: file '/Users/runner/.cache/bazel/install/055c20a5e[6](https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/3441965197/jobs/5762053586#step:4:7)23d48b44a7c0f17b28c93a/A-server.jar' is missing or modified.  Please remove '/Users/runner/.cache/bazel/install/055c20a5e623d48b44a7c0f17b28c93a' and try again.
> Error: Process completed with exit code 36.
> ```
> 
> can be done as explained [here](https://stackoverflow.com/a/64819132/10933809)

Thanks @esigo .

**Pull request merged**